src/pages/Login.tsx

```diff
@@ -81,9 +82,40 @@ const LoginPage: React.FC<LoginPageProps> = ({ newLogin }) => {
     }
   }, [selectedProfile]);
 
+  const handleSubmit = (values: ProfileFormValues) => {
+    void sessionOps.signIn(values.serverAddress, {
+      credentials: {
+        password: {
+          name: values.name,
+          password: values.password,
+          persistLogin: values.persistLogin,
+        },
+      },
+    });
+  };
+
   if (session.loading) {
-    return <LoadingPlaceholder />;
+    if (session.loading && session.databaseState) {
+      return (
+        <LandingPageError>
+          <ProfileForm
+            onSubmit={handleSubmit}
+            error="Unable to establish a connection with the database... Try again."
+            secondaryUri="/login"
+            secondaryLabel="Create profile"
+            submitLabel="Log in"
+            defaultValues={{
+              name: selectedProfile?.name,
+              serverAddress: selectedProfile?.serverAddress,
+            }}
+          />
+        </LandingPageError>
+      );
+    } else {
+      return <LoadingPlaceholder />;
+    }
   }
```

Comment:
duplicating this code here is hacky... want to keep the elements in the dom to allow some sort of transition consider toggling class name modifiers on the existing element... we don't do transitions like this anywhere else though so i would rather not - we will eventually take another pass at the ux to improve things like this across the app consistently